jgrapht-core/src/main/java/org/jgrapht/alg/KDisjointShortestPaths.java

```diff
@@ -0,0 +1,387 @@
+ * 
+ * @see BellmanFordShortestPath
+ *
+ * @author Assaf Mizrachi
+ * @since September 11, 2016
+ */
+public class KDisjointShortestPaths<V, E> {
+	/**
+	 * Graph on which shortest paths are searched.
+	 */
+	private Graph<V, E> graph;
+
+	private List<List<E>> pathList;
+
+	private int nPaths;
+
+	private V startVertex;
+
+	/**
+	 * Creates an object to calculate k disjoint shortest paths between the start
+	 * vertex and others vertices.
+	 *
+	 * @param graph
+	 *            graph on which shortest paths are searched.
+	 * @param startVertex
+	 *            start vertex of the calculated paths.
+	 * @param nPaths
+	 *            number of disjoint paths between the start vertex and an end
+	 *            vertex.
+	 *
+	 * @throws NullPointerException
+	 *             if the specified graph or startVertex is <code>null</code>.
+	 * @throws IllegalArgumentException
+	 *             if nPaths is negative or 0.
+	 */
+	public KDisjointShortestPaths(Graph<V, E> graph, V startVertex, int nPaths) {
+		assertKShortestPathsFinder(graph, startVertex, nPaths);
+
+		this.graph = graph;
+		this.startVertex = startVertex;
+		this.nPaths = nPaths;
+	}
+
+	/**
+	 * Returns the k shortest simple paths from start vertex to end vertex
+	 * in increasing order of weight.
+	 *
+	 * @param endVertex
+	 *            target vertex of the calculated disjoint paths.
+	 *
+	 * @return list of paths, or <code>null</code> if no path exists between the
+	 *         start vertex and the end vertex.
+	 */
+	public List<GraphPath<V, E>> getPaths(V endVertex) {
+		assertGetPaths(endVertex);
+		//ensure we have directed weighted graph
+		assert(this.graph instanceof DirectedGraph<?, ?>);
+		this.graph = new AsWeightedDirectedGraph<V, E>((DirectedGraph<V, E>) graph, new HashMap<>());
+
+		List<E> currentPath;
+		this.pathList = new ArrayList<>();
+		BellmanFordShortestPath<V, E> bellmanFordShortestPath;
+		int cPaths = 1;
+		do {
+			setUp(cPaths);
+			bellmanFordShortestPath = new BellmanFordShortestPath<V, E>(this.graph, this.startVertex);
+			currentPath = bellmanFordShortestPath.getPathEdgeList(endVertex);
+			if (currentPath != null) {
+				cPaths++;				
+				pathList.add(currentPath);
+			}			
+		} while (currentPath != null && cPaths <= this.nPaths);
+
+		return pathList.size() > 0 ? tearDown(endVertex) : null;
+	}
+
+	/**
+	 * Prepares the graph for a search of the next path:
+	 * Replacing the edges of the previous path with reversed edges
+	 * with negative weight
+	 * 
+	 * @param cPath the number of the next path to search 
+	 */
+	private void setUp(int cPath) {
+		//no setup for first path
+		if (cPath == 1) {
+			return;
+		}
+		
+		V source, target;
+		E reversedEdge;
+		//replace previous path edges with reversed edges with negative weight
+		for (E originalEdge : this.pathList.get(cPath - 2)) {
+			source = graph.getEdgeSource(originalEdge);
+			target = graph.getEdgeTarget(originalEdge);
+			graph.removeEdge(originalEdge);	

```

Comment:
I understand these are two different comments here. For the first (modifying the original graph), I think the solution should be to add a warning on that at the comments and not that this class to provide a deep copy solution of the original graph, it's out of its scope. The second comment - not if the provided graph is a simple directed graph which is what the algorithm requires.